### PR TITLE
feat: collect `BLSToExecutionChange` from gossip

### PR DESCRIPTION
**Pull request by MegaRedHand**
Related to #918

### CODE REVIEW

File: lib/lambda_ethereum_consensus/fork_choice/fork_choice.ex [hunk 1]
```diff
@@ -166,6 +167,7 @@ defmodule LambdaEthereumConsensus.ForkChoice do
 
     Handlers.notify_forkchoice_update(store, head_block)
 
+    OperationsCollector.notify_new_block(head_block)
```
Comment (mpaulucci) [inline lib/lambda_ethereum_consensus/fork_choice/fork_choice.ex:170]: nice! Seems this `recompute_head` has grown its responsabilities, basically doing every task that is async. We may want to do a refactor soon (out of scope of this PR)

File: lib/lambda_ethereum_consensus/p2p/gossip/handler.ex [hunk 2]
```diff
@@ -31,7 +39,7 @@ defmodule LambdaEthereumConsensus.P2P.Gossip.Handler do
     root = aggregate.data.beacon_block_root |> Base.encode16()
 
     # We are getting ~500 attestations in half a second. This is overwhelming the store GenServer at the moment.
-    # Store.on_attestation(aggregate)
+    # ForkChoice.on_attestation(aggregate)
```
Comment (mpaulucci) [inline lib/lambda_ethereum_consensus/p2p/gossip/handler.ex:42]: Oh, I didn't know this was still commented out. Created: https://github.com/lambdaclass/lambda_ethereum_consensus/issues/924

File: lib/lambda_ethereum_consensus/p2p/gossip/operations_collector.ex [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+      operations.bls_to_execution_changes
+      |> MapSet.new(& &1.message.validator_index)
+
+    bls_to_execution_changes =
+      state.bls_to_execution_change
+      |> Enum.reject(&MapSet.member?(indices, &1.message.validator_index))
+
+    %{state | bls_to_execution_change: bls_to_execution_changes}
+  end
+end
```
Comment (mpaulucci) [inline lib/lambda_ethereum_consensus/p2p/gossip/operations_collector.ex:49]: we need a getter here as well for the validator to query, right?

Comment (mpaulucci) [inline lib/lambda_ethereum_consensus/p2p/gossip/operations_collector.ex:49]: I like this design. I guess we want to also be able to remove these as slots/epochs pass by. What do you think of Beacon Chain emiting an event every time a slot (or epoch) passes and genservers that are interesed in slot changes subscribing to that event? (again, probably out of scope of this PR)

File: lib/lambda_ethereum_consensus/fork_choice/fork_choice.ex (hunk 1 above)
Comment (MegaRedHand) [reply lib/lambda_ethereum_consensus/fork_choice/fork_choice.ex:170]: 👍 #925

File: lib/lambda_ethereum_consensus/p2p/gossip/operations_collector.ex (hunk 3 above)
Comment (MegaRedHand) [reply lib/lambda_ethereum_consensus/p2p/gossip/operations_collector.ex:49]: I was thinking the `Validator` needs something like that (but for every third of a slot). I think it's something we want to tackle (although maybe after we finish the `Validator`)

**Review by mpaulucci — APPROVED** [review]

File: lib/lambda_ethereum_consensus/p2p/gossip/operations_collector.ex (hunk 3 above)
Comment (MegaRedHand) [reply lib/lambda_ethereum_consensus/p2p/gossip/operations_collector.ex:49]: #926

**Comment by MegaRedHand** [comment]
@mpaulucci I'm gonna add the getter before merging

File: lib/lambda_ethereum_consensus/p2p/gossip/operations_collector.ex (hunk 3 above)
Comment (MegaRedHand) [reply lib/lambda_ethereum_consensus/p2p/gossip/operations_collector.ex:61]: > I guess we want to also be able to remove these as slots/epochs pass by.

Actually, this kind of operation should be valid until it's added in a block (or a similar one is).

**Pull request merged**